src/main/kotlin/com/feedback/resource/FeedbackResource.kt

```diff
@@ -0,0 +1,52 @@
+package com.feedback.resource
+
+import com.feedback.dao.FeedbackCreationRequest
+import com.feedback.dao.LoggedInUser
+import com.feedback.service.EmployeeFetcher
+import com.feedback.service.FeedbackInserter
+import jakarta.ws.rs.Consumes
+import jakarta.ws.rs.POST
+import jakarta.ws.rs.Path
+import jakarta.ws.rs.Produces
+import jakarta.ws.rs.core.Context
+import jakarta.ws.rs.core.Cookie
+import jakarta.ws.rs.core.Response
+import jakarta.ws.rs.core.Response.Status
+import org.springframework.stereotype.Component
+
+@Component
+@Path("/api/feedback")
+@Produces("application/json")
+@Consumes("application/json")
+class FeedbackResource(
+    private val feedbackInserter: FeedbackInserter,
+    private val employeeFetcher: EmployeeFetcher,
+    private val loginDetailValidator: LoginDetailValidator,
+    private val extractCookieData: ExtractCookieData
+) {
+
+    @POST
+    fun submitFeedback(
+        feedbackRequest: FeedbackRequest,
+        @Context cookies: Map<String, Cookie>
+    ): Response {
+        val loggedInUser = extractCookieData.extractCompanyIdEmployeeId(cookies)
+
+        val department = fetchEmployeeDepartmentByAnonymity(loggedInUser, feedbackRequest.isAnonymous)
+        val feedbackCreationRequest = FeedbackCreationRequest(department, feedbackRequest.comment)
+
+        loginDetailValidator.validateCompanyId(loggedInUser.companyId)
+        loginDetailValidator.validateEmployeeId(loggedInUser.employeeId)
+
+        feedbackInserter.insertFeedback(loggedInUser, feedbackCreationRequest)
+
```

Comment:
we want to keep the Resource (Controller) as simple as possible. So after you extract the loggedIn user, create a new place where you do this validation + db handling. 
We don't want to create Test for this file, but this file won't be "stupid" and will have logic, we have a problem.

Reply:
I'm not sure I understand your last sentence - "create Test for this file" => the Resource I believe, while do testing  the new class...?

Reply:
changed a bit, but currently I'm using validator class, and then:
```
val department = fetchEmployeeDepartmentByAnonymity(loggedInUser, feedbackRequest.isAnonymous)
val feedbackCreationRequest = FeedbackCreationRequest(department, feedbackRequest.comment)
```
so should I create new class for fetching by anonymity?

Reply:
what I'm saying is, that the Resource should only authenticate the user and then delegate the data to the "service", or whatever logic. If you put some calls other than authentication, then the Resource isn't "stupid" anymore and then you need to provide a test for it. Since I don't want to write any tests for Resource, I expect to extract logic to other classes